### ISSUE TITLE
[Question] How to use ReactiveValidationObject and INotifyDataErrorInfo in WPF

### ISSUE THREAD

**Issue by lszczygielek** [reactiveui/ReactiveUI.Validation#237]
Hi,
I'm playing with validation using `INotifyDataErrorInfo` and I'm very confused and I don't know how to use it properly. I couldn't find concrete info in documentation, only sample project gives me some clues, but I don't know if it is all what I should know.

Let's say that I want to make a `TextBox` and default ErrorTemplate (red border) should be visible if value is empty. Based on code from sample I'm suspecting that I have to:
- bind `TextBox` `Text` to view-model using regular XAML binding,
- in the code behind of view I have to bind using ReactiveUI a `ViewModel` to `DataContext`.

So in short - is there another way to use `INotifyDataErrorInfo` using only ReactiveUi bindings? Unfortunately I can't turn on `INotifyDataErrorInfo` in other way than regular XAML binding, documentation didn't help me in this topic, only sample application shows me how to do this in regular XAML.

**Comment by ScarletKuro** [comment]
> Is there another way to use INotifyDataErrorInfo using only ReactiveUi bindings?

No. The INotifyDataErrorInfo is only supported via XAML binding, it doesn't support BindValidation.

If i remember correct, the reason is because RxUI has it's own binding mechanism.
I also think there was idea to make platform depended libs like ReactiveUI.Validation.WPF to support it, but im not sure.

**Comment by glennawatson** [comment]
We listen and do the setting of the property ourselves in RxUI bindings, and don't use the inbuilt classes that WPF provides, it's more multi platform aware that way.

The downside is that INotifyDataErrorInfo relies on WPF inbuilt binding classes.

**Comment by glennawatson** [comment]
There's nothing stopping you from using the inbuilt WPF Bind with RXUI, we abstract it out into a `ICreatesObservable` interface so you could make sure of those classes behind the scenes.

Might be a future WPF/UWP package feature.

**Comment by lszczygielek** [comment]
Thank you all for clarification.

@glennawatson what do you think about to add info that `INotifyDataErrorInfo`  requires classic binding/connecting to build-in WPF mechanism? I can make PR with it but first I would like to know if this makes sense.

**Comment by lszczygielek** [comment]
Documentation updated, so I'm closing this. Thank you all for your time in clarification and PR review.
